client/app/services/query.js

```diff
@@ -293,6 +294,21 @@ function QueryResource($resource, $http, $q, $location, currentUser, QueryResult
     return this.getParameters().get();
   };
 
+  Query.prototype.getLocalParametersDefs = function getLocalParametersDefs() {
+    if (!this.$localParameters) {
+      this.$localParameters = this.getParametersDefs().filter(p => !p.global);
```

Comment:
Normal parameters aren't cached -- whenever we call `Parameters.get` it calls `Parameters.updateParameters` (which is a no-op in case the query text didn't change). Or are we talking about different things?